storage/operation/pebbleimpl/writer.go

```diff
@@ -40,6 +45,13 @@ func (b *ReaderBatchWriter) PebbleWriterBatch() *pebble.Batch {
 	return b.batch
 }
 
+// Lock tries to acquire the lock for the batch.
+// if the lock is already acquired by this same batch from other pending db operations,
+// then it will not be blocked and can continue updating the batch, which prevents a re-entrant deadlock.
```

Comment:
```suggestion
// then it will not be blocked and can continue updating the batch, which prevents a re-entrant deadlock.
// CAUTION: The caller must ensure that no other references exist for the input lock.
```